core/src/main/java/greencity/controller/EcoNewsCommentController.java

```diff
@@ -0,0 +1,224 @@
+package greencity.controller;
+
+import greencity.annotations.ApiPageable;
+import greencity.constant.HttpStatuses;
+import greencity.dto.PageableDto;
+import greencity.dto.comment.CommentReturnDto;
+import greencity.dto.econewscomment.EcoNewsCommentDto;
+import greencity.dto.econewscomment.AddEcoNewsCommentDtoRequest;
+import greencity.entity.User;
+import greencity.service.EcoNewsCommentService;
+import greencity.service.EcoNewsService;
+import greencity.service.UserService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.security.Principal;
+import java.util.List;
+import javax.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import springfox.documentation.annotations.ApiIgnore;
+
+@Validated
+@AllArgsConstructor
+@RestController
+@RequestMapping("/econews/comments")
+public class EcoNewsCommentController {
+    private final EcoNewsCommentService ecoNewsCommentService;
+    private final UserService userService;
+    private final EcoNewsService ecoNewsService;
+
+    /**
+     * Method for creating {@link greencity.entity.EcoNewsComment}.
+     *
+     * @param econewsId id of {@link greencity.entity.EcoNews} to add comment to.
+     * @param addEcoNewsCommentDtoRequest - dto for {@link greencity.entity.EcoNewsComment} entity.
+     * @return dto {@link greencity.dto.econewscomment.AddEcoNewsCommentDtoResponse}
+     */
+    @ApiOperation(value = "Add comment.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = HttpStatuses.CREATED, response = CommentReturnDto.class),
+        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @PostMapping("{econewsId}")
+    public ResponseEntity<Object> save(@PathVariable Long econewsId,
```

Comment:
- Why `Object` is used as generic type?
- Why method is called `save`, javaDoc mentions that it should create smth, the response code is CREATED, but all the method really does is `findByEmail`?

Reply:
> * Why `Object` is used as generic type?
 It will be fixed in next pull requests
> * Why method is called `save`, javaDoc mentions that it should create smth, the response code is CREATED, but all the method really does is `findByEmail`?
 The main logic of method is located in return statement : .body(ecoNewsCommentService.save(econewsId, addEcoNewsCommentDtoRequest, user))